src/index.js

```diff
@@ -170,7 +170,7 @@ class MiniCssExtractPlugin {
                   renderedModules,
                   compilation.runtimeTemplate.requestShortener
                 ),
-              filenameTemplate: this.options.filename,
+              filenameTemplate: this.getFilename(chunk, this.options.filename),
```

Comment:
I believe you can pass a function to `filenameTemplate` too. webpack will call it with an information object.